avcodec.cpp

```diff
@@ -116,22 +116,20 @@ bool avcodec_decoder_is_streamable(const opencv_mat mat) {
     const cv::Mat* buf = static_cast<const cv::Mat*>(mat);
     size_t bufSize = buf->total();
 
-    while (bytesRead < probeBytesLimit && bytesRead < static_cast<int64_t>(bufSize)) {
-        // Check if there's enough buffer left for another atom header
-        if (bufSize - bytesRead < atomHeaderSize) {
-            break;
-        }
-
+    while (bytesRead + atomHeaderSize <= probeBytesLimit && bytesRead + atomHeaderSize <= static_cast<int64_t>(bufSize)) {
         // Read atom size and type
         uint32_t atomSize = (buf->data[bytesRead] << 24) | (buf->data[bytesRead + 1] << 16) |
                             (buf->data[bytesRead + 2] << 8) | buf->data[bytesRead + 3];
 
+        // Validate atom size
+        if (atomSize < atomHeaderSize || atomSize + bytesRead > bufSize || atomSize > probeBytesLimit - bytesRead) {
```

Comment:
i think that `atomSize > probeBytesLimit - bytesRead` shouldn't be checked here -- i guess it's ok if the atom is larger than the `probeBytesLimit`, as long as we don't read it. I think the check for probe limit in the loop is sufficient

Reply:
great point, yeah, this check is redundant and can be removed.